TTS/tts/multispeaker/adverserial_classifier.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from torch.nn import functional as F
+from torch.nn import Sequential, Linear
+
+
+class GradientReversalFunction(torch.autograd.Function):
+    """Revert gradient without any further input modification."""
+
+    @staticmethod
+    def forward(ctx, x, l, c):
+        ctx.l = l
+        ctx.c = c
+        return x.view_as(x)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        grad_output = grad_output.clamp(-ctx.c, ctx.c)
+        return ctx.l * grad_output.neg(), None, None
+
+class ReversalClassifier(torch.nn.Module):
+    """Adversarial classifier (with two FC layers) with a gradient reversal layer.
+
+    Arguments:
+        input_dim -- size of the input layer (probably should match the output size of encoder)
+        hidden_dim -- size of the hiden layer
+        output_dim -- number of channels of the output (probably should match the number of speakers/languages)
+        gradient_clipping_bound (float) -- maximal value of the gradient which flows from this module
+    Keyword arguments:
+        scale_factor (float, default: 1.0)-- scale multiplier of the reversed gradientts
+    """
+
+    def __init__(self, input_dim, hidden_dim, output_dim, gradient_clipping_bounds, scale_factor=1.0):
+        super(ReversalClassifier, self).__init__()
+        self._lambda = scale_factor
+        self._clipping = gradient_clipping_bounds
+        self._output_dim = output_dim
+        self._classifier = Sequential(
+            Linear(input_dim, hidden_dim),
+            Linear(hidden_dim, output_dim)
+        )
+
+    def forward(self, x):
+        x = GradientReversalFunction.apply(x, self._lambda, self._clipping)
+        x = self._classifier(x)
+        return x
+
+    @staticmethod
+    def loss(input_lengths, speakers, prediction):
```

Comment:
To follow the style it might be interesting to add this loss function to TTS/tts/layers/losses.py
Does it make sense to you? Since the other models/modules, the loss function is in TTS/tts/layers/losses.py

Reply:
will do :)